test/jdk/java/text/testlib/HexDumpReader.java

```diff
@@ -41,80 +41,30 @@
     }
 
+    // Converts a Hex dump file (File) into an InputStream containing bytes
     public static InputStream getStreamFromHexDump(File hexFile) {
-        ByteArrayBuilder bab = new ByteArrayBuilder();
-        int lineNo = 0;
-        try (BufferedReader reader
-                 = new BufferedReader(new InputStreamReader(new FileInputStream(hexFile),
-                StandardCharsets.US_ASCII))) {
-            String line;
-            while ((line = reader.readLine()) != null) {
-                lineNo++;
-                line = line.trim();
-                // Skip blank and comment lines.
-                if (line.length() == 0) {
-                    continue;
-                }
-                int x = line.indexOf('#');
-                if (x == 0) {
-                    continue;
-                }
-                if (x > 0) {
-                    line = line.substring(0, x).trim();
-                }
-                int len = line.length();
-                for (int i = 0; i < len; i += 2) {
-                    bab.put((byte)Integer.parseInt(line, i, i + 2, 16));
-                }
-            }
-        } catch (Exception e) {
-            throw new RuntimeException(hexFile.getName() + ":error:" + lineNo + ": " + e, e);
-        }
-        return new ByteArrayInputStream(bab.toArray());
-    }
-
-
-    private static class ByteArrayBuilder {
-        private static final int BUFFER_SIZE = 4096;
-
-        private int size;
-        private List<byte[]> bytes;
-        private byte[] current;
-        private int offset;
-
-        ByteArrayBuilder() {
-            bytes = new ArrayList<>();
-            current = new byte[BUFFER_SIZE];
-        }
-
-        void put(byte b) {
-            if (offset == BUFFER_SIZE) {
-                bytes.add(current);
-                current = new byte[BUFFER_SIZE];
-                offset = 0;
-            }
-            current[offset++] = b;
-            size++;
-        }
-
-        byte[] toArray() {
-            byte[] buf = new byte[size];
-            int ptr = 0;
-            for (byte[] ba : bytes) {
-                System.arraycopy(ba, 0, buf, ptr, ba.length);
-                ptr += ba.length;
-            }
-            System.arraycopy(current, 0, buf, ptr, offset);
-            assert ptr + offset == size;
-            return buf;
+        List<String> lines;
+        try {
+            lines = Files.readAllLines(hexFile.toPath(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new RuntimeException(String.format("Utility class HexDumpReader" +
+                    " threw %s when trying to read the file %s", e, hexFile.getName()));
         }
+        // Grab all non-empty lines up until a '#' if one exists
+        String hexString = lines.stream().map(String::trim)
+                .map(s -> (s.contains("#")) ? s.substring(0, s.indexOf("#")).trim() : s)
+                .filter(s -> !s.isEmpty()).collect(Collectors.joining());
```

Comment:
Thanks for the review. Talked with Lance offline, and since none of the existing dump files actually use a non-leading comment, I changed the stream to only look for leading comments (removed ternary map operation).

As a result, I adjusted the documentation to state that a valid valid hex dump file format does not allow non-leading comments.